platform-hub-api/app/controllers/concerns/kubernetes_groups_sub_collection.rb

```diff
@@ -0,0 +1,18 @@
+module KubernetesGroupsSubCollection
+  extend ActiveSupport::Concern
+
+  # `resource` is expected to have a `#kubernetes_groups` association
+  def kubernetes_groups_sub_collection resource, target = nil
+    scope = resource.kubernetes_groups.not_privileged
```

Comment:
If `kubernetes_groups` is not available on the resource, then it's a design/code time error, not a runtime error, so probably better to let this hard crash and catch this in dev/testing.

(I guess this the typical duck typing approach we do in Ruby? Otherwise we'd be checking using `respond_to?` in many places).

Reply:
👍